### PR TITLE
Update patches for 1.8.0

### PR DESCRIPTION
**Pull request by jjyao**
Added a `ray-data` subpackage corresponding to ray["data"].

Removed `ray-autoscaler` and `ray-debug` since there is no corresponding extra.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by h-vetinari** [comment]
Thanks for getting started on this! Patches don't yet apply cleanly...

**Comment by jjyao** [comment]
@h-vetinari @krfricke @vnlitvinov It's ready for review.

**Review by h-vetinari — APPROVED** [review]
Can you comment on the removal of ray-autoscaler and ray-debug? Are those not supported upstream anymore?

Otherwise LGTM, thanks!

**Comment by jjyao** [comment]
Yea, there is no corresponding upstream ray["autoscaler"] and ray["debug"]. And they were not really enabled in conda before.

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
  if setup_spec.type == SetupType.RAY:
      setup_spec.extras = {
+-        "data": [
+-            "pandas",
+-            "pyarrow>=4.0.1",
+-            "fsspec",
+-        ],
 -        "default": [
--            "aiohttp",
+-            "aiohttp >= 3.7",
```
Comment (vnlitvinov) [inline recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:25]: this change should be reflected in `meta.yaml` dependencies of `ray-core`

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
 -            "aiohttp_cors",
 -            "aioredis < 2",
 -            "colorful",
 -            "py-spy >= 0.2.0",
--            "jsonschema",
 -            "requests",
--            "gpustat",
+-            "gpustat >= 1.0.0b1",  # for windows
```
Comment (vnlitvinov) [inline recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:31]: this, too

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch [hunk 3]
```diff
@@ -72,6 +77,7 @@ index 7ebc0a894..f70018f27 100644
 -        "dataclasses; python_version < '3.7'",
 -        "filelock",
 -        "grpcio >= 1.28.1",
+-        "jsonschema",
```
Comment (vnlitvinov) [inline recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:80]: and this, too

**Review by krfricke — APPROVED** [review]
Looks good, thanks!

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch (hunk 3 above)
Comment (vnlitvinov) [reply recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:80]: oh, this one is already reflected

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch (hunk 2 above)
Comment (jjyao) [reply recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:31]: Unfortunately, there is only 0.6 version of gpustat in conda-forge, which is already specified in `meta.yaml`.

File: recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch (hunk 1 above)
Comment (jjyao) [reply recipe/patches/0004-Remove-all-dependencies-from-setup.py.patch:25]: Fixed.

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**